src/ILCompiler.Compiler/src/Compiler/CompilerGeneratedInteropStubManager.cs

```diff
@@ -136,6 +136,31 @@ yield return
                     };
             }
         }
+        
+        private void AddDependenciesDueToPInvokeStructDelegateField(ref DependencyList dependencies, NodeFactory factory, TypeDesc typeDesc)
+        {
+            if (typeDesc is ByRefType)
+            {
+                typeDesc = typeDesc.GetParameterType();
+            }
+
+            MetadataType metadataType = typeDesc as MetadataType;
+            if (metadataType != null) 
+            {
+                foreach (FieldDesc field in metadataType.GetFields())
+                {
+                    if (field.IsStatic)
+                    {
+                        continue;
+                    }
+
+                    if (field.FieldType.IsDelegate)
+                    {
+                        AddDependenciesDueToPInvokeDelegate(ref dependencies, factory, field.FieldType);
```

Comment:
Do we need to handle embedded structs here?

```
struct A {
    MyDelegate d1;
}

struct B { // <- this should generate marshaling code for MyDelegate
    A a; 
}
```

Reply:
Yes we need to handle the embedded struct scenario.